RucioException crashes when kwargs are passed

### Description

https://github.com/rucio/rucio/blob/5d42f2c54bc60a213701bb6b77bf810835a280b4/lib/rucio/common/exception.py#L26-L54

The issue is with this line:

https://github.com/rucio/rucio/blob/5d42f2c54bc60a213701bb6b77bf810835a280b4/lib/rucio/common/exception.py#L34

`Exception` does not take `kwargs`. Passing a keyword argument to `RucioException` thus results in:

```
    def __init__(self, *args, **kwargs):
>       super(RucioException, self).__init__(*args, **kwargs)
E       TypeError: MockException() takes no keyword arguments
```



### Steps to reproduce

Pass a kwarg to `RucioException` or to one of its subclasses

### Rucio Version

_No response_

### Additional Information

_No response_

--- a/lib/rucio/common/exception.py
+++ b/lib/rucio/common/exception.py
@@ -31,7 +31,7 @@ class RucioException(Exception):
     """
 
     def __init__(self, *args, **kwargs):
-        super(RucioException, self).__init__(*args, **kwargs)
+        super(RucioException, self).__init__(*args)
         self._message = "An unknown exception occurred."
         self.args = args
         self.kwargs = kwargs

--- a/lib/rucio/common/exception.py
+++ b/lib/rucio/common/exception.py
@@ -30,20 +30,15 @@ class RucioException(Exception):
     with the keyword arguments provided to the constructor.
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args):
         super(RucioException, self).__init__(*args)
         self._message = "An unknown exception occurred."
         self.args = args
-        self.kwargs = kwargs
         self.error_code = 1
         self._error_string = None
 
     def __str__(self):
-        try:
-            self._error_string = self._message % self.kwargs
-        except Exception:
-            # at least get the core message out if something happened
-            self._error_string = self._message
+        self._error_string = self._message
         if len(self.args) > 0:
             # If there is a non-kwarg parameter, assume it's the error
             # message or reason description and tack it on to the end
@@ -60,8 +55,8 @@ class AccessDenied(RucioException):
     """
     AccessDenied
     """
-    def __init__(self, *args, **kwargs):
-        super(AccessDenied, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(AccessDenied, self).__init__(*args)
         self._message = "Access to the requested resource denied."
         self.error_code = 2
 
@@ -70,8 +65,8 @@ class AccountNotFound(RucioException):
     """
     AccountNotFound
     """
-    def __init__(self, *args, **kwargs):
-        super(AccountNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(AccountNotFound, self).__init__(*args)
         self._message = "Account does not exist."
         self.error_code = 3
 
@@ -80,8 +75,8 @@ class CannotAuthenticate(RucioException):
     """
     CannotAuthenticate
     """
-    def __init__(self, *args, **kwargs):
-        super(CannotAuthenticate, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(CannotAuthenticate, self).__init__(*args)
         self._message = "Cannot authenticate."
         self.error_code = 4
 
@@ -90,8 +85,8 @@ class ClientParameterMismatch(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ClientParameterMismatch, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ClientParameterMismatch, self).__init__(*args)
         self._message = "Client parameters don\'t match."
         self.error_code = 5
 
@@ -100,8 +95,8 @@ class ClientProtocolNotSupported(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ClientProtocolNotSupported, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ClientProtocolNotSupported, self).__init__(*args)
         self._message = "Client protocol not supported."
         self.error_code = 6
 
@@ -110,8 +105,8 @@ class ConfigNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ConfigNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ConfigNotFound, self).__init__(*args)
         self._message = "Configuration not found."
         self.error_code = 7
 
@@ -120,8 +115,8 @@ class ConfigurationError(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ConfigurationError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ConfigurationError, self).__init__(*args)
         self._message = "Error during configuration."
         self.error_code = 8
 
@@ -130,8 +125,8 @@ class CounterNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(CounterNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(CounterNotFound, self).__init__(*args)
         self._message = "The requested counter does not exist."
         self.error_code = 9
 
@@ -140,8 +135,8 @@ class DatabaseException(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(DatabaseException, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DatabaseException, self).__init__(*args)
         self._message = "Database exception."
         self.error_code = 10
 
@@ -150,8 +145,8 @@ class DataIdentifierAlreadyExists(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(DataIdentifierAlreadyExists, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DataIdentifierAlreadyExists, self).__init__(*args)
         self._message = "Data Identifier Already Exists."
         self.error_code = 11
 
@@ -160,8 +155,8 @@ class DataIdentifierNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(DataIdentifierNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DataIdentifierNotFound, self).__init__(*args)
         self._message = "Data identifier not found."
         self.error_code = 12
 
@@ -170,8 +165,8 @@ class DestinationNotAccessible(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(DestinationNotAccessible, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DestinationNotAccessible, self).__init__(*args)
         self._message = "Access to local destination denied."
         self.error_code = 13
 
@@ -180,8 +175,8 @@ class Duplicate(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(Duplicate, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(Duplicate, self).__init__(*args)
         self._message = "An object with the same identifier already exists."
         self.error_code = 14
 
@@ -190,8 +185,8 @@ class DuplicateContent(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(DuplicateContent, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DuplicateContent, self).__init__(*args)
         self._message = "Data identifier already added to the destination content."
         self.error_code = 15
 
@@ -200,8 +195,8 @@ class DuplicateRule(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(DuplicateRule, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DuplicateRule, self).__init__(*args)
         self._message = "A duplicate rule for this account, did, rse_expression, copies already exists."
         self.error_code = 16
 
@@ -210,8 +205,8 @@ class ErrorLoadingCredentials(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ErrorLoadingCredentials, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ErrorLoadingCredentials, self).__init__(*args)
         self._message = "Unable to to load user credentials."
         self.error_code = 17
 
@@ -220,8 +215,8 @@ class FileAlreadyExists(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(FileAlreadyExists, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(FileAlreadyExists, self).__init__(*args)
         self._message = "The file already exists."
         self.error_code = 18
 
@@ -230,8 +225,8 @@ class FileConsistencyMismatch(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(FileConsistencyMismatch, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(FileConsistencyMismatch, self).__init__(*args)
         self._message = "Error related to file consistency."
         self.error_code = 19
 
@@ -240,8 +235,8 @@ class FileReplicaAlreadyExists(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(FileReplicaAlreadyExists, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(FileReplicaAlreadyExists, self).__init__(*args)
         self._message = "File name in specified scope already exists"
         self.error_code = 20
 
@@ -250,8 +245,8 @@ class ReplicaNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ReplicaNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ReplicaNotFound, self).__init__(*args)
         self._message = "Replica not found"
         self.error_code = 21
 
@@ -260,8 +255,8 @@ class ReplicaUnAvailable(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ReplicaUnAvailable, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ReplicaUnAvailable, self).__init__(*args)
         self._message = "Replica unavailable"
         self.error_code = 22
 
@@ -270,8 +265,8 @@ class FullStorage(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(FullStorage, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(FullStorage, self).__init__(*args)
         self._message = "The Referenced storage is out of disk space."
         self.error_code = 23
 
@@ -280,15 +275,15 @@ class IdentityError(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(IdentityError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(IdentityError, self).__init__(*args)
         self._message = "Identity error."
         self.error_code = 24
 
 
 class IdentityNotFound(RucioException):
-    def __init__(self, *args, **kwargs):
-        super(IdentityNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(IdentityNotFound, self).__init__(*args)
         self._message = "This identity does not exist."
         self.error_code = 25
 
@@ -297,8 +292,8 @@ class InputValidationError(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InputValidationError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InputValidationError, self).__init__(*args)
         self._message = "There is an error with one of the input parameters."
         self.error_code = 26
 
@@ -307,8 +302,8 @@ class InsufficientAccountLimit(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InsufficientAccountLimit, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InsufficientAccountLimit, self).__init__(*args)
         self._message = "There is not enough quota left to fulfil the operation."
         self.error_code = 27
 
@@ -317,8 +312,8 @@ class InsufficientTargetRSEs(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InsufficientTargetRSEs, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InsufficientTargetRSEs, self).__init__(*args)
         self._message = "There are not enough target RSEs to fulfil the request at this time."
         self.error_code = 28
 
@@ -327,8 +322,8 @@ class InvalidMetadata(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidMetadata, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidMetadata, self).__init__(*args)
         self._message = "Provided metadata is considered invalid."
         self.error_code = 29
 
@@ -337,8 +332,8 @@ class InvalidObject(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidObject, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidObject, self).__init__(*args)
         self._message = "Provided object does not match schema."
         self.error_code = 30
 
@@ -347,8 +342,8 @@ class InvalidReplicationRule(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidReplicationRule, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidReplicationRule, self).__init__(*args)
         self._message = "Provided replication rule is considered invalid."
         self.error_code = 31
 
@@ -357,8 +352,8 @@ class InvalidRSEExpression(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidRSEExpression, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidRSEExpression, self).__init__(*args)
         self._message = "Provided RSE expression is considered invalid."
         self.error_code = 32
 
@@ -367,8 +362,8 @@ class InvalidRuleWeight(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidRuleWeight, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidRuleWeight, self).__init__(*args)
         self._message = "An invalid weight value/type is used for an RSE."
         self.error_code = 33
 
@@ -377,8 +372,8 @@ class InvalidType(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidType, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidType, self).__init__(*args)
         self._message = "Provided type is considered invalid."
         self.error_code = 34
 
@@ -387,8 +382,8 @@ class InvalidValueForKey(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidValueForKey, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidValueForKey, self).__init__(*args)
         self._message = "Invalid value for the key."
         self.error_code = 35
 
@@ -397,8 +392,8 @@ class InvalidRequest(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidRequest, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidRequest, self).__init__(*args)
         self._message = "Request is considered invalid."
         self.error_code = 36
 
@@ -407,8 +402,8 @@ class InvalidPath(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(InvalidPath, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidPath, self).__init__(*args)
         self._message = "The path provided is invalid."
         self.error_code = 37
 
@@ -417,8 +412,8 @@ class KeyNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(KeyNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(KeyNotFound, self).__init__(*args)
         self._message = "Key does not exist."
         self.error_code = 38
 
@@ -427,8 +422,8 @@ class LifetimeExceptionDuplicate(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(LifetimeExceptionDuplicate, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(LifetimeExceptionDuplicate, self).__init__(*args)
         self._message = "An exception already exists."
         self.error_code = 39
 
@@ -437,8 +432,8 @@ class LifetimeExceptionNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(LifetimeExceptionNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(LifetimeExceptionNotFound, self).__init__(*args)
         self._message = "Exception does not exist."
         self.error_code = 40
 
@@ -447,8 +442,8 @@ class ManualRuleApprovalBlocked(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ManualRuleApprovalBlocked, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ManualRuleApprovalBlocked, self).__init__(*args)
         self._message = "Manual rule approval is blocked on this RSE."
         self.error_code = 41
 
@@ -457,8 +452,8 @@ class MissingClientParameter(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(MissingClientParameter, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(MissingClientParameter, self).__init__(*args)
         self._message = "Client parameters are missing."
         self.error_code = 42
 
@@ -467,8 +462,8 @@ class MissingDependency(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(MissingDependency, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(MissingDependency, self).__init__(*args)
         self._message = "One dependency is missing."
         self.error_code = 43
 
@@ -477,8 +472,8 @@ class MissingSourceReplica(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(MissingSourceReplica, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(MissingSourceReplica, self).__init__(*args)
         self._message = "Source replicas are missing to fulfil the request at this moment."
         self.error_code = 44
 
@@ -487,8 +482,8 @@ class NameTypeError(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(NameTypeError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NameTypeError, self).__init__(*args)
         self._message = "Name is of the wrong type"
         self.error_code = 45
 
@@ -497,8 +492,8 @@ class NoAuthInformation(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(NoAuthInformation, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NoAuthInformation, self).__init__(*args)
         self._message = "No authentication information passed."
         self.error_code = 46
 
@@ -507,8 +502,8 @@ class NoFilesDownloaded(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(NoFilesDownloaded, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NoFilesDownloaded, self).__init__(*args)
         self._message = "None of the requested files have been downloaded."
         self.error_code = 75
 
@@ -517,8 +512,8 @@ class NotAllFilesDownloaded(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(NotAllFilesDownloaded, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NotAllFilesDownloaded, self).__init__(*args)
         self._message = "Not all of the requested files have been downloaded."
         self.error_code = 76
 
@@ -527,15 +522,15 @@ class ReplicationRuleCreationTemporaryFailed(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ReplicationRuleCreationTemporaryFailed, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ReplicationRuleCreationTemporaryFailed, self).__init__(*args)
         self._message = "The creation of the replication rule failed at this time. Please try again later."
         self.error_code = 47
 
 
 class RequestNotFound(RucioException):
-    def __init__(self, *args, **kwargs):
-        super(RequestNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RequestNotFound, self).__init__(*args)
         self._message = "A request for this DID and RSE does not exist."
         self.error_code = 48
 
@@ -544,8 +539,8 @@ class RSEAccessDenied(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEAccessDenied, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEAccessDenied, self).__init__(*args)
         self._message = "Referenced RSE not reachable."
         self.error_code = 49
 
@@ -554,8 +549,8 @@ class RSEWriteBlocked(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEWriteBlocked, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEWriteBlocked, self).__init__(*args)
         self._message = "RSE excluded; not available for writing."
         self.error_code = 50
 
@@ -564,8 +559,8 @@ class RSENotConnected(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSENotConnected, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSENotConnected, self).__init__(*args)
         self._message = "Connection to RSE not established."
         self.error_code = 51
 
@@ -574,8 +569,8 @@ class RSENotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSENotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSENotFound, self).__init__(*args)
         self._message = "RSE does not exist."
         self.error_code = 52
 
@@ -584,8 +579,8 @@ class RSEProtocolNotSupported(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEProtocolNotSupported, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEProtocolNotSupported, self).__init__(*args)
         self._message = "RSE does not support requested protocol."
         self.error_code = 53
 
@@ -594,8 +589,8 @@ class RSEProtocolPriorityError(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEProtocolPriorityError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEProtocolPriorityError, self).__init__(*args)
         self._message = "RSE does not support provided protocol priority for protocol."
         self.error_code = 54
 
@@ -604,8 +599,8 @@ class RSEProtocolDomainNotSupported(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEProtocolDomainNotSupported, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEProtocolDomainNotSupported, self).__init__(*args)
         self._message = "RSE does not support requested protocol scope."
         self.error_code = 55
 
@@ -614,8 +609,8 @@ class RSEOperationNotSupported(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEOperationNotSupported, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEOperationNotSupported, self).__init__(*args)
         self._message = "RSE does not support requested operation."
         self.error_code = 56
 
@@ -624,8 +619,8 @@ class RSEFileNameNotSupported(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEFileNameNotSupported, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEFileNameNotSupported, self).__init__(*args)
         self._message = "RSE does not support provided filename."
         self.error_code = 57
 
@@ -634,8 +629,8 @@ class RSEOverQuota(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEOverQuota, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEOverQuota, self).__init__(*args)
         self._message = "Quota of Referenced RSE is exceeded."
         self.error_code = 58
 
@@ -644,8 +639,8 @@ class ResourceTemporaryUnavailable(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ResourceTemporaryUnavailable, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ResourceTemporaryUnavailable, self).__init__(*args)
         self._message = "The resource is temporary not available."
         self.error_code = 59
 
@@ -654,8 +649,8 @@ class RuleNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RuleNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RuleNotFound, self).__init__(*args)
         self._message = "No replication rule found."
         self.error_code = 60
 
@@ -664,8 +659,8 @@ class RuleReplaceFailed(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(RuleReplaceFailed, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RuleReplaceFailed, self).__init__(*args)
         self._message = "The replace operation for the rule failed."
         self.error_code = 61
 
@@ -674,8 +669,8 @@ class ScratchDiskLifetimeConflict(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ScratchDiskLifetimeConflict, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ScratchDiskLifetimeConflict, self).__init__(*args)
         self._message = "The requested replication rule exceeds the maximum SCRATCHDISK lifetime of 15 days."
         self.error_code = 62
 
@@ -684,8 +679,8 @@ class ServiceUnavailable(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ServiceUnavailable, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ServiceUnavailable, self).__init__(*args)
         self._message = "The requested service is not available at the moment."
         self.error_code = 63
 
@@ -694,8 +689,8 @@ class ScopeAccessDenied(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ScopeAccessDenied, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ScopeAccessDenied, self).__init__(*args)
         self._message = "Access to Referenced scope denied."
         self.error_code = 64
 
@@ -704,8 +699,8 @@ class ScopeNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ScopeNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ScopeNotFound, self).__init__(*args)
         self._message = "Scope does not exist."
         self.error_code = 65
 
@@ -714,8 +709,8 @@ class SourceAccessDenied(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(SourceAccessDenied, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(SourceAccessDenied, self).__init__(*args)
         self._message = "Access to local source file denied."
         self.error_code = 66
 
@@ -724,8 +719,8 @@ class SourceNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(SourceNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(SourceNotFound, self).__init__(*args)
         self._message = "Source file not found."
         self.error_code = 67
 
@@ -734,8 +729,8 @@ class StagingAreaRuleRequiresLifetime(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(StagingAreaRuleRequiresLifetime, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(StagingAreaRuleRequiresLifetime, self).__init__(*args)
         self._message = "A rule involving a staging area requires a lifetime!"
         self.error_code = 68
 
@@ -744,8 +739,8 @@ class SubscriptionDuplicate(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(SubscriptionDuplicate, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(SubscriptionDuplicate, self).__init__(*args)
         self._message = "A subscription with the same identifier already exists."
         self.error_code = 69
 
@@ -754,8 +749,8 @@ class SubscriptionNotFound(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(SubscriptionNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(SubscriptionNotFound, self).__init__(*args)
         self._message = "Subscription not found."
         self.error_code = 70
 
@@ -764,8 +759,8 @@ class UnsupportedDIDType(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedDIDType, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedDIDType, self).__init__(*args)
         self._message = "Unsupported DID type for this operation. Only DATASET or FILE is allowed."
         self.error_code = 71
 
@@ -774,8 +769,8 @@ class UnsupportedOperation(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedOperation, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedOperation, self).__init__(*args)
         self._message = "The resource doesn't support the requested operation."
         self.error_code = 72
 
@@ -784,8 +779,8 @@ class UnsupportedStatus(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedStatus, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedStatus, self).__init__(*args)
         self._message = "Unsupported data identifier status."
         self.error_code = 73
 
@@ -794,8 +789,8 @@ class UnsupportedValueType(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedValueType, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedValueType, self).__init__(*args)
         self._message = "Unsupported type for the value. List of supported types: %s." % str(AUTHORIZED_VALUE_TYPES)
         self.error_code = 74
 
@@ -804,8 +799,8 @@ class MissingModuleException(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(MissingModuleException, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(MissingModuleException, self).__init__(*args)
         self._message = "The module is not installed."
         self.error_code = 77
 
@@ -814,8 +809,8 @@ class ServerConnectionException(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(ServerConnectionException, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ServerConnectionException, self).__init__(*args)
         self._message = "Cannot connect to the Rucio server."
         self.error_code = 78
 
@@ -824,8 +819,8 @@ class NoFilesUploaded(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(NoFilesUploaded, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NoFilesUploaded, self).__init__(*args)
         self._message = "None of the given files have been uploaded."
         self.error_code = 79
 
@@ -834,8 +829,8 @@ class NotAllFilesUploaded(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(NotAllFilesUploaded, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NotAllFilesUploaded, self).__init__(*args)
         self._message = "Not all of the given files have been uploaded."
         self.error_code = 80
 
@@ -844,8 +839,8 @@ class RSEChecksumUnavailable(RucioException):
     """
     Cannot retrieve checksum from RSE
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEChecksumUnavailable, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEChecksumUnavailable, self).__init__(*args)
         self._message = "RSE checksum unavailable."
         self.error_code = 81
 
@@ -854,8 +849,8 @@ class UndefinedPolicy(RucioException):
     """
     Cannot find a defined policy in the Rucio config
     """
-    def __init__(self, *args, **kwargs):
-        super(UndefinedPolicy, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UndefinedPolicy, self).__init__(*args)
         self._message = "No policy is defined."
         self.error_code = 82
 
@@ -864,8 +859,8 @@ class TransferToolTimeout(RucioException):
     """
     Timeout from the transfer tool
     """
-    def __init__(self, *args, **kwargs):
-        super(TransferToolTimeout, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(TransferToolTimeout, self).__init__(*args)
         self._message = "Timeout from the transfer tool."
         self.error_code = 83
 
@@ -874,8 +869,8 @@ class TransferToolWrongAnswer(RucioException):
     """
     Wrong answer returned by the transfer tool
     """
-    def __init__(self, *args, **kwargs):
-        super(TransferToolWrongAnswer, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(TransferToolWrongAnswer, self).__init__(*args)
         self._message = "Wrong answer returned by the transfer tool."
         self.error_code = 84
 
@@ -884,8 +879,8 @@ class RSEAttributeNotFound(RucioException):
     """
     RSE attribute not found.
     """
-    def __init__(self, *args, **kwargs):
-        super(RSEAttributeNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(RSEAttributeNotFound, self).__init__(*args)
         self._message = "RSE attribute not found."
         self.error_code = 85
 
@@ -894,8 +889,8 @@ class UnsupportedKeyType(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedKeyType, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedKeyType, self).__init__(*args)
         self._message = "Unsupported type for the key."
         self.error_code = 86
 
@@ -904,8 +899,8 @@ class MetalinkJsonParsingError(RucioException):
     """
     Failed to parse input with metalink and json
     """
-    def __init__(self, data, metalink_err, json_err, *args, **kwargs):
-        super(MetalinkJsonParsingError, self).__init__(*args, **kwargs)
+    def __init__(self, data, metalink_err, json_err, *args):
+        super(MetalinkJsonParsingError, self).__init__(*args)
         self._message = 'Failed parsing of %s. MetalinkError: %s. JsonError: %s' % (data, metalink_err, json_err)
         self.error_code = 87
 
@@ -914,8 +909,8 @@ class ReplicaIsLocked(RucioException):
     """
     Replica has one or more locks.
     """
-    def __init__(self, *args, **kwargs):
-        super(ReplicaIsLocked, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ReplicaIsLocked, self).__init__(*args)
         self._message = 'Replica is locked'
         self.error_code = 88
 
@@ -924,8 +919,8 @@ class UnsupportedRequestedContentType(RucioException):
     """
     The requested content type is not supported by the API endpoint.
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedRequestedContentType, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedRequestedContentType, self).__init__(*args)
         self._message = 'The requested content type is not supported.'
         self.error_code = 89
 
@@ -934,8 +929,8 @@ class DuplicateFileTransferSubmission(RucioException):
     """
     A transfer for the same file is already submitted to the Transfer Tool.
     """
-    def __init__(self, *args, **kwargs):
-        super(DuplicateFileTransferSubmission, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DuplicateFileTransferSubmission, self).__init__(*args)
         self._message = 'One or more files are already submitted to the transfer tool'
         self.error_code = 90
 
@@ -944,8 +939,8 @@ class DIDError(RucioException):
     """
     An operation related to DID type went wrong
     """
-    def __init__(self, *args, **kwargs):
-        super(DIDError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DIDError, self).__init__(*args)
         self._message = 'Error using DID type'
         self.error_code = 91
 
@@ -954,8 +949,8 @@ class NoDistance(RucioException):
     """
     No distance can be found between 2 RSEs
     """
-    def __init__(self, *args, **kwargs):
-        super(NoDistance, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(NoDistance, self).__init__(*args)
         self._message = 'Cannot found a distance between 2 RSEs'
         self.error_code = 92
 
@@ -964,8 +959,8 @@ class PolicyPackageNotFound(RucioException):
     """
     The policy package specified in the config file was not found
     """
-    def __init__(self, *args, **kwargs):
-        super(PolicyPackageNotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(PolicyPackageNotFound, self).__init__(*args)
         self._message = 'The specified policy package was not found'
         self.error_code = 93
 
@@ -974,8 +969,8 @@ class CannotAuthorize(RucioException):
     """
     Failed to authorize an operation.
     """
-    def __init__(self, *args, **kwargs):
-        super(CannotAuthorize, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(CannotAuthorize, self).__init__(*args)
         self._message = 'Can not authorize operation.'
         self.error_code = 94
 
@@ -984,8 +979,8 @@ class SubscriptionWrongParameter(RucioException):
     """
     RucioException
     """
-    def __init__(self, *args, **kwargs):
-        super(SubscriptionWrongParameter, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(SubscriptionWrongParameter, self).__init__(*args)
         self._message = "Subscription wrong parameters"
         self.error_code = 95
 
@@ -994,8 +989,8 @@ class VONotFound(RucioException):
     """
     Requested VO does not exist.
     """
-    def __init__(self, *args, **kwargs):
-        super(VONotFound, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(VONotFound, self).__init__(*args)
         self._message = 'The requested VO does not exist'
         self.error_code = 96
 
@@ -1004,8 +999,8 @@ class UnsupportedAccountName(RucioException):
     """
     Requested account name is not supported for users.
     """
-    def __init__(self, *args, **kwargs):
-        super(UnsupportedAccountName, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(UnsupportedAccountName, self).__init__(*args)
         self._message = 'The requested account name cannot be used'
         self.error_code = 97
 
@@ -1014,8 +1009,8 @@ class DuplicateCriteriaInDIDFilter(RucioException):
     """
     Duplicate criteria found in DID filter.
     """
-    def __init__(self, *args, **kwargs):
-        super(DuplicateCriteriaInDIDFilter, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DuplicateCriteriaInDIDFilter, self).__init__(*args)
         self._message = 'Duplicate criteria for key/operator in filter expression: {}'.format(args[0])
         self.error_code = 98
 
@@ -1024,8 +1019,8 @@ class DIDFilterSyntaxError(RucioException):
     """
     DID filter is not parsable.
     """
-    def __init__(self, *args, **kwargs):
-        super(DIDFilterSyntaxError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DIDFilterSyntaxError, self).__init__(*args)
         self._message = 'Syntax error in filter expression.'
         self.error_code = 99
 
@@ -1034,8 +1029,8 @@ class InvalidAlgorithmName(RucioException):
     """
     The given algorithm name is not valid for the VO.
     """
-    def __init__(self, algorithm, vo, *args, **kwargs):
-        super(InvalidAlgorithmName, self).__init__(*args, **kwargs)
+    def __init__(self, algorithm, vo, *args):
+        super(InvalidAlgorithmName, self).__init__(*args)
         self.message = 'Algorithm name %s is not valid for VO %s' % (algorithm, vo)
         self.error_code = 100
 
@@ -1044,8 +1039,8 @@ class FilterEngineGenericError(RucioException):
     """
     Generic Filter Engine error.
     """
-    def __init__(self, *args, **kwargs):
-        super(FilterEngineGenericError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(FilterEngineGenericError, self).__init__(*args)
         self._message = 'Generic filter engine error.'
         self.error_code = 101
 
@@ -1054,8 +1049,8 @@ class MetadataSchemaMismatchError(RucioException):
     """
     External table does not match expected table schema.
     """
-    def __init__(self, *args, **kwargs):
-        super(MetadataSchemaMismatchError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(MetadataSchemaMismatchError, self).__init__(*args)
         self._message = 'The external table does not match the expected table schema.'
         self.error_code = 102
 
@@ -1064,8 +1059,8 @@ class PolicyPackageVersionError(RucioException):
     """
     Policy package is not compatible with this version of Rucio.
     """
-    def __init__(self, package, *args, **kwargs):
-        super(PolicyPackageVersionError, self).__init__(*args, **kwargs)
+    def __init__(self, package, *args):
+        super(PolicyPackageVersionError, self).__init__(*args)
         self._message = 'Policy package %s is not compatible with this Rucio version' % package
         self.error_code = 103
 
@@ -1075,8 +1070,8 @@ class InvalidSourceReplicaExpression(RucioException):
     Source Replica Expression Considered Invalid
     """
 
-    def __init__(self, *args, **kwargs):
-        super(InvalidSourceReplicaExpression, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(InvalidSourceReplicaExpression, self).__init__(*args)
         self._message = 'Provided Source Replica expression is considered invalid.'
         self.error_code = 104
 
@@ -1085,8 +1080,8 @@ class DeprecationError(RucioException):
     """
     Function has been deprecated.
     """
-    def __init__(self, *args, **kwargs):
-        super(DeprecationError, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(DeprecationError, self).__init__(*args)
         self._message = 'Command or function has been deprecated.'
         self.error_code = 105
 
@@ -1095,8 +1090,8 @@ class SortingAlgorithmNotSupported(RucioException):
     """
     Sorting algorithm is not supported.
     """
-    def __init__(self, *args, **kwargs):
-        super(SortingAlgorithmNotSupported, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(SortingAlgorithmNotSupported, self).__init__(*args)
         self._message = 'Sorting algorithm is not supported.'
         self.error_code = 106
 
@@ -1105,7 +1100,7 @@ class ErrorLoadingPolicyPackage(RucioException):
     """
     An error occurred while loading the policy package.
     """
-    def __init__(self, *args, **kwargs):
-        super(ErrorLoadingPolicyPackage, self).__init__(*args, **kwargs)
+    def __init__(self, *args):
+        super(ErrorLoadingPolicyPackage, self).__init__(*args)
         self._message = 'An error occurred while loading the specified policy package'
         self.error_code = 107

--- a/tests/rucio/common/test_exception.py
+++ b/tests/rucio/common/test_exception.py
@@ -1,0 +1,29 @@
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from rucio.common.exception import RucioException
+
+
+class TestRucioException:
+    class MockException(RucioException):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self._message = "Test exception."
+
+    def test_rucio_exception_str(self):
+        exc = self.MockException()
+        assert str(exc) == "Test exception."
+
+    def test_rucio_exception_str_extra_error_message(self):
+        exc = self.MockException("Extra message.", "Second extra message.")
+        assert str(exc) == "Test exception.\nDetails: Extra message.\nSecond extra message."

